### DISCUSSION TITLE
STILL HAPPENING -> Invalid copilot token missing token 403

### DISCUSSION THREAD

**Discussion by silostack** [community/community#43390] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

I understand this issue seems to have a LOT of threads already but the threads seem to have been addressed (or at least it started working for people), but I JUST setup copilot this morning and I'm getting this with every IntelliJ product and Visual Code. Super frustrating onboarding experience....

**Comment by Akash1134** [comment, marked as answer]
Hi @silostack , Sorry that you're getting the "Missing Token 403" error while using Copilot with both IntelliJ and Visual Code. If you've already tried the steps mentioned in this [discussion](https://github.com/community/community/discussions/40019) and it hasn't helped, I'd recommend getting in touch with our Support team for a thorough investigation by opening a ticket on our [Support page](https://support.github.com/contact). Please open a ticket with them :)

**Reply by silostack** [reply]
Got in touch w/support. I had to fix the Copilot settings here: https://github.com/settings/copilot. Super weird cause I'm pretty sure I had check that yesterday but... 🤷‍♂️ Anyway, working now!

**Reply by d-apps** [reply]
> Got in touch w/support. I had to fix the Copilot settings here: https://github.com/settings/copilot. Super weird cause I'm pretty sure I had check that yesterday but... 🤷‍♂️ Anyway, working now!

This fixed my issue.

**Comment by alibaizhanov** [comment]
STILL HAPPENING -> Invalid copilot token missing token 403

**Comment by alibaizhanov** [comment]
STILL HAPPENING -> Invalid copilot token missing token 403
<img width="608" alt="image" src="https://user-images.githubusercontent.com/32927647/213543189-e1f0fb16-7e4d-4f71-9da5-551f6f3d6f87.png">

**Reply by davecheney** [reply]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Comment by almazmusic** [comment]
Same thing. Log out didn't help. Created a support ticket.

**Comment by ilya-lysko** [comment]
Same thing is happening now

**Comment by MrAndriy** [comment]
Same thing too. Log out didn't help. Created a support ticket

**Comment by shap-po** [comment]
Same problem here
- sign out & sign in ❌
- reinstall ❌
- switch to Nightly ❌

**Comment by OleksiiZubko** [comment]
Same thing too. Log out didn't help.

**Comment by riZZZhik** [comment]
Looks like GitHub removed Copilot from "Student Developer Pack offers"
I have used Copilot since release with this pack, now in the profile settings (https://github.com/settings/copilot) it shows that I don't have license

**Reply by riZZZhik** [reply]
<img width="1124" alt="Screenshot 2023-02-16 at 2 08 43 AM" src="https://user-images.githubusercontent.com/21087403/219211105-c06f8ad1-4b29-44e1-abb7-60d36716f4bd.png">

**Reply by shredder67** [reply]
When I registered about half a month ago, this page also didn't have information on student pack. I think you should check your **Billing and plans** page, there should be a section on Copilot being active (I do, and I use student pack too).

Edit: added screenshot
![image](https://user-images.githubusercontent.com/31409973/219214244-0f6bddb7-3792-471a-9ecd-35aa58b2f19c.png)

**Reply by riZZZhik** [reply]
Same there

<img width="789" alt="Screenshot 2023-02-16 at 2 18 29 AM" src="https://user-images.githubusercontent.com/21087403/219214172-71eb1c3e-0f96-4e50-81c3-f3c7ef10e03b.png">

**Reply by riZZZhik** [reply]
New error:

```
[INFO] [auth] [2023-02-17T08:47:54.690Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2023-02-17T08:47:54.692Z] GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"
```

**Reply by riZZZhik** [reply]
Fixed in Profile Settings, but still `"At this time, Copilot is not available in your location"` in editors

![Screenshot 2023-02-20 at 4 05 28 PM](https://user-images.githubusercontent.com/21087403/220116952-2d012f46-eb59-4b31-a6c9-9296ef9ff9e3.png)

![Screenshot 2023-02-20 at 4 05 40 PM](https://user-images.githubusercontent.com/21087403/220116941-f851b913-209e-472b-bbd2-a091b9a80334.png)

**Comment by shredder67** [comment]
Stopped working about 3 hours ago, tried everything from thread (logging out and in, reinstalled extension, checked profile settings and billing), nothing helps

**Comment by max-trix** [comment]
Same. Where should I get in touch with support? Is that github support or another?

**Comment by innokentiy2517** [comment]
Same issue with token again. Last month just writing in thread helped. Should we now contact support or just wait for the fix?

**Comment by zifelok** [comment]
Same issue both VS and VS Code

**Reply by zifelok** [reply]
Even after logout&login

**Reply by zifelok** [reply]
Works fine today

**Comment by dabeedj** [comment]
Same here with PhpStorm

**Reply by vitalik1921** [reply]
The same here VSCode

**Comment by ShmatkoAndrey** [comment]
Same problem

**Comment by SechinAndrey** [comment]
**Update: it works for me now**

Same problem VS Code

**Comment by fahad0samara** [comment]
Same problem

**Comment by mihoaka** [comment]
same problem arrgh

**Comment by Bohooslav** [comment]
Same issue for me. It stopped working this morning, so I bought a yearly subscription. But it doesn't work. Logged out then logged back in, reinstall, nothing works.

**Comment by brutalsavagerekt** [comment]
Same for me. Invalid copilot token: missing token: 403

**Comment by Helyk** [comment]
I don't see this exact output with `Copilot could not connect to server` posed by anyone else so I'll drop it here. Also, tried everything mentioned plus reinstalling Copilot (both regular and Nightly).
```
[INFO] [auth] [2023-02-16T05:17:46.841Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2023-02-16T05:17:46.846Z] GitHub Copilot could not connect to server. Extension activation failed: "Contact Support"
```

**Comment by noname-oni** [comment]
🇺🇦 💛💙
Anyone from Ukraine, come to the neighboring discussion: [discussions/47577](https://github.com/orgs/community/discussions/47577)

**Comment by d-apps** [comment]
I was having this issue and this fixed:

Go to **Settings** -> **Copilot**.
In **Suggestions matching public code** select **Allow**.
And mark **Allow GitHub to use my code snippets for product improvements**
Click **Save**.
Try to login again.

![Screen Shot 2023-02-16 at 08 26 37](https://user-images.githubusercontent.com/20110433/219353012-b207119d-56f1-4920-9d7d-bc2cb732dbc8.png)

![Screen Shot 2023-02-16 at 08 26 55](https://user-images.githubusercontent.com/20110433/219353022-85f67883-5a69-48c4-9f46-4be4847ec7dc.png)

**Reply by max-trix** [reply]
What's your location?

**Reply by d-apps** [reply]
Brazil, in the second I did this the login was successful.

**Reply by max-trix** [reply]
Thanks

**Reply by Helyk** [reply]
Tried that, didn't help.

**Comment by shredder67** [comment]
I think it's save to say that most people having problems with using Copilot are from RU and UA. I doubt that Microsoft would enable some policies without making an announcement beforehand, so I hope an issue will be fixed (or at least addressed).

**Comment by dabeedj** [comment]
Keep russia banned. Thank you for fixing Ukraine

**Comment by aimanuslim** [comment]
From Malaysia here, still getting 403. I am playing my hard earned money for this and getting such poor service.

**Comment by abdalazizrashid** [comment]
Same here!

**Comment by Akash1134** [comment]
Hi everyone, 

If you're getting the error 403 while using Copilot, a bug fix has been deployed to production and affected users should be able to connect.

Due to the recency of this incident, we are still investigating the contributing factors and will provide a more detailed update on cause and remediation in the February Availability Report, which we will publish the first Wednesday of March.

We sincerely apologize for any stress and/or frustration this has caused.

**Reply by riZZZhik** [reply]
Hi,

What about "At this time, Copilot is not available in your location" without any official announcement?
Keeping this answer in mind: https://github.com/orgs/community/discussions/12042#discussioncomment-2276190
